src/main/java/com/mael/payroll/PayrollDBFacade.java

```diff
@@ -30,6 +32,43 @@ public void deleteEmployee(int employeeId) {
         return listOfEmployees;
     }
 
+    public void addUnionMember(int memberId, int employeeId) {
+        this.unionMembers.put(memberId, employeeId);
+    }
+
+    public Employee getUnionMember(int memberId) {
+        if (employeeExistsAndIsMember(memberId)) {
+            return getEmployee(unionMembers.get(memberId));
+        }
+        throw new UnionMemberNotInDBException();
```

Comment:
Good catch. I was probably tired.

- If there is no member with that id, `unionMembers.get(memberId)` would return null, and `getEmployee(null)` would raise an `EmployeeNotInDBException`

- If the member id exists, but there is no employee with an id of `unionMembers.get(memberId)`, then again, `getEmployee(smoke)` would raise an `EmployeeNotInDBException`.

Which makes me think, that there should be no item in `unionMembers` holding an employee id that doesn't exist anymore... that would mean that when I use the `DeleteEmployeeTransaction`, I should check too if the employee is part of any union and remove their member id from the hashmap...

The question is, if you fire an employee, could it still be part of the union? Or is the union not independent of the company?

For now, I will just return `getEmployee(unionMembers.get(memberId))`